### DISCUSSION TITLE
Get a field from another table through a relationship

### DISCUSSION THREAD

**Discussion by cnrodrig** [Budibase/budibase#10317] [Help] [answered]
Hi!

I'm working with a Postgres database, and i have two tables: Request and Product. There is a relationship between the tables (defined in budibase) where 1 request can have only one product. 

In a request/:id screen where i can see a particular request, i have a button that execute a REST post to an endpoint. 

The endpoint receives data from the request and also some data (fields) from the product table.

In the binding of the REST API call, i'm trying, unsuccessfully to access the related data.

Things i have tried, for example, to get the name of the product:

DataProvider.Rows.0.product.0.name
DataProvider.Rows.0.product.name
DataProvider.Rows.product.0.name

Here DataProvider is binded to the Request table, product is the name of the relationship field as defined in the data section and name is the field in the product table.

Can you help me?

**Comment by melohagan** [comment, marked as answer]
Hey @cnrodrig 

You cannot access relational data directly through the parent Data provider binding like that. You need to nest a repeater and nest another data provider.
This is because relationships are always stored in an array (even if there's just a single relationship)

An example of accessing relational data is here: https://docs.budibase.com/docs/fetching-data#displaying-relational-data

So the simplest example setup would be:

> Repeater block (Parent table -> filtered to display only one entry)
> > Data provider / block (Relationship of the parent table)

![Screenshot 2023-04-18 at 14 05 48](https://user-images.githubusercontent.com/101575380/232786562-71a8471c-9b72-4b5b-bf14-2bf9435c3233.png)
